execution/registry/cache.go

```diff
@@ -106,22 +107,22 @@ func (cache *Cache) RemoveNode(addr crypto.Address) error {
 func (cache *Cache) Sync(state Writer) error {
 	cache.Lock()
 	defer cache.Unlock()
-	var addresses []crypto.Address
-	for addr := range cache.registry {
-		addresses = append(addresses, addr)
+	var ids []crypto.Address
+	for id := range cache.registry {
+		ids = append(ids, id)
 	}
 
-	for _, addr := range addresses {
-		info := cache.registry[addr]
+	for _, id := range ids {
+		info := cache.registry[id]
```

Comment:
Here you're building an array just to loop over it. Could it not be simplified with:
for info, id := range cache.registry {
}
and do away with ids?
note that if you know the size of an array, best to allocate it with that capicity to reprevent reallocs:
ids := make([]crypto.Address, 0, len(cache.registry))
however I don't think ids is needed.